bitnami/postgresql/templates/_helpers.tpl

```diff
@@ -160,7 +160,7 @@ Return true if a secret object should be created
 */}}
 {{- define "postgresql.createSecret" -}}
 {{- $customUser := include "postgresql.username" . -}}
-{{- $postgresPassword := include "common.secrets.lookup" (dict "secret" (include "common.names.fullname" .) "key" .Values.auth.secretKeys.adminPasswordKey "defaultValue" (ternary (coalesce .Values.global.postgresql.auth.postgresPassword .Values.auth.postgresPassword) (coalesce .Values.global.postgresql.auth.password .Values.auth.password) (or (empty $customUser) (eq $customUser "postgres"))) "context" $) -}}
+{{- $postgresPassword := include "common.secrets.lookup" (dict "secret" (include "common.names.fullname" .) "key" .Values.auth.secretKeys.adminPasswordKey "defaultValue" (ternary (coalesce .Values.global.postgresql.auth.password .Values.auth.password .Values.global.postgresql.auth.postgresPassword .Values.auth.postgresPassword) (coalesce .Values.global.postgresql.auth.postgresPassword .Values.auth.postgresPassword) (or (empty $customUser) (eq $customUser "postgres"))) "context" $) -}}
```

Comment:
I do not understand why we are changing this. 

- If customUser is empty or if it's postgres, we should use .Values.global.postgresql.auth.postgresPassword or .Values.auth.postgresPassword that are the parameters for that
- If the user is using a custom user, he should be setting a password in the correct parameter: .Values.global.postgresql.auth.password .Values.auth.password

Reply:
Let me explain in detail:

`$postgresPassword` stores the password of the 'postgres' user.

We want this value to be set to:
- If 'username' is empty or equals 'postgres', the password for Postgres user can be set using any of the following, in this order: `.Values.global.postgresql.auth.password`, `.Values.auth.password`, `.Values.global.postgresql.auth.postgresPassword`, `.Values.auth.postgresPassword`.
- If 'username' is not empty and not equals 'postgres' (in other words, it is set), the password for Postgres user can be set using any of the following, in this order: `.Values.global.postgresql.auth.postgresPassword`, `.Values.auth.postgresPassword`.

The currently logic prior to this PR is completely wrong, were if `auth.username` is set and password `auth.password` is provided, the Postgres user will also use that password.

Reply:
In that case, let's update the values/README to because we say that the parameter is for the custom user to create 

https://github.com/bitnami/charts/blob/main/bitnami/postgresql/values.yaml#L23

Reply:
I don't think it is needed, as the current description still suits the behavior, this is simply to allow the following use cases:
- ```yaml
  auth:
    enablePostgresUser: true
    postgresPassword: "my_pass"
    username: ""
    password: ""
  ```
- ```yaml
  auth:
    enablePostgresUser: true
    postgresPassword: ""
    username: "postgres"
    password: "my_pass"
  ```

It simply, to avoid an even more complex logic, also allows the following settings which may feel a bit weird:
- ```yaml
  auth:
    enablePostgresUser: true
    postgresPassword: ""
    username: ""
    password: "my_pass"
  ```
- ```yaml
  auth:
    enablePostgresUser: true
    postgresPassword: "my_pass"
    username: "postgres"
    password: ""
   ```

Please notice this logic is inherited from the Postgresql container logic which also allows this behavior: https://github.com/bitnami/containers/blob/8384822f1fcaea2991ad5e64849a046c7d9671ae/bitnami/postgresql/15/debian-11/rootfs/opt/bitnami/scripts/libpostgresql.sh#L641-L648